sdlib/d/gc/hpd.d

```diff
@@ -121,10 +121,14 @@ public:
 		auto start = allocatedPages.findSetBackward(index) + 1;
 		auto stop = allocatedPages.findSet(index + pages - 1);
 
-		allocCount--;
 		usedCount -= pages;
 		longestFreeRange = max(longestFreeRange, stop - start);
 	}
+
+	void release(uint index, uint pages) {
+		shrinkAlloc(index, pages);
+		allocCount--;
+	}
```

Comment:
You got to love how simple this became. But it wasn't obvious it could be that simple from the get go. First, shrink and release had to have their common part factored. Then in turn it made it obvious that shrink's API was inadequate, and as layers of complexity are being removed, you end up with something that's so simple it's obviously correct, won't contain bugs and will be easy to maintain.